src/custom/pages/Claim/index.tsx

```diff
@@ -146,11 +141,28 @@ export default function Claim() {
           setClaimStatus(ClaimStatus.DEFAULT)
           console.log(error)
         })
+    }
+
+    // check if there are any selected (paid) claims
+    let inputData
+    if (!selected.length) {
+      inputData = freeClaims.map(({ index }) => ({ index }))
+      console.log('Starting claiming with', inputData)
+      sendTransaction(inputData)
+    } else if (investFlowStep == 2) {
+      // Free claimings + selected investment oportunities
+      const selectedIndex = [...getIndexes(freeClaims), ...selected]
+      inputData = selectedIndex.reduce<EnhancedUserClaimData[]>((acc, idx: number) => {
```

Comment:
i changed this into a reduce instead of a map + filter, because TS was giving me a hard time. It was thinking the array of clamings could contain undefined when i was filtering them...
reduce worked :)

Reply:
should we break apart this fn?

Reply:
u mean the reduce? we could, but not sure it would add a lot of value. 
if we had time, would be nice to cleanup things a bit, cause the logic still is a bit hard to follow